src/modules/qSpecies/process.cpp

```diff
@@ -17,19 +18,50 @@ Module::ExecutionResult QSpeciesModule::process(ModuleContext &moduleContext)
         return ExecutionResult::Failed;
     }
 
-    // Ensure any parameters in our nodes are set correctly
-    if (excludeSameMolecule_)
-        selectNF_->setSameMoleculeExclusions({selectBO_});
-    else
-        selectNF_->setSameMoleculeExclusions({});
-    selectNF_->keywords().set("InclusiveRange", distanceRange_);
+    // Select all potential bridging oxygen sites - we will determine which actually are
+    // involved in NF-BO-NF interactions once we have the available NF sites
+    SiteSelector allOxygenSites(targetConfiguration_, bridgingOxygenSpeciesSites_);
 
-    // Execute the analysis
-    if (!analyser_.execute({moduleContext.dissolve(), targetConfiguration_, name()}))
+    // Select all NF centres
+    SiteSelector NF(targetConfiguration_, networkFormerSpeciesSites_);
+
+    // Filter the oxygen sites into those surrounded by exactly two NF sites
+    SiteFilter filter(targetConfiguration_, allOxygenSites.sites());
+    auto &&[BO, neighbourMap] = filter.filterBySiteProximity(NF.sites(), distanceRange_, 0, 2);
+
+    // The returned 'neighbourMap' maps BO sites to nearby NF sites *only if* there were exactly two NF sites within range.
+    // So, we can use this to determine the Q numbers for each NF by counting the number of times a NF site appears in the map.
+    std::map<const Site *, int> qSpecies;
+    std::map<int, int> oxygenTypes;
+    for (const auto &[siteBO, nbrNF] : neighbourMap)
     {
-        Messenger::error("Q Species experienced problems with its analysis.\n");
-        return ExecutionResult::Failed;
+        for (const auto &[nbr, nbrIndex] : nbrNF)
+        {
+            ++oxygenTypes[nbrNF.size()];
+            if (nbrNF.size() == 2)
+                ++qSpecies[nbr];
+        }
```

Comment:
@Danbr4d Careful!  As written this will increase your `oxygenTypes` counting for every NF site defined for each oxygen. So, you want to move the increment and the size check outside of the loop
```suggestion
        ++oxygenTypes[nbrNF.size()];
        if (nbrNF.size() == 2)
        {
            for (const auto &[nbr, nbrIndex] : nbrNF)
            {
                    ++qSpecies[nbr];
            }
        }
```